### PR TITLE
RD can emag mech fabricator(like roboticist).

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request
Mech fabricator now can be emaged by user who have cyborg skillchip(that one that show cyborg wires).
RD and roboticist have this skillchip.

## Why It's Good For The Game

RD is like roboticist, but better.

## Changelog

:cl:
add: RD can emag mech fabricator(like roboticist).
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
wait, what does an emagged mech fabricator do?

**Comment by zoomachina** [comment]
> wait, what does an emagged mech fabricator do?

It unlocks Justice mech

**Comment by zxaber** [comment]
Why is emagging an exofab locked behind jobs at all?

**Comment by necromanceranne** [comment]
Have it tied to the robotics skillchip's trait. ``TRAIT_KNOW_ROBO_WIRES``

No need to make it roundstart job specific.

File: code/modules/vehicles/mecha/mech_fabricator.dm [hunk 1]
```diff
@@ -149,7 +149,7 @@
 /obj/machinery/mecha_part_fabricator/emag_act(mob/user, obj/item/card/emag/emag_card)
 	if(obj_flags & EMAGGED)
 		return FALSE
-	if(user.job != JOB_ROBOTICIST)
+	if(!(user.job == JOB_ROBOTICIST || user.job == JOB_RESEARCH_DIRECTOR))
```
Comment (Iamgoofball) [inline code/modules/vehicles/mecha/mech_fabricator.dm:152]: Why does this even have a job lock on it anyways? No other emag act has a job lock like this.

Comment (kawaiinick) [reply code/modules/vehicles/mecha/mech_fabricator.dm:152]: it was requested by oranges

**Comment by Jacquerel** [comment]
> Have it tied to the robotics skillchip's trait. `TRAIT_KNOW_ROBO_WIRES`
> 
> No need to have it tied to job datum.

Do this preferably

**Comment by Xackii** [comment]
made as TRAIT_KNOW_ROBO_WIRES check

File: code/modules/vehicles/mecha/mech_fabricator.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/modules/vehicles/mecha/mech_fabricator.dm]: Where did he request this?

File: code/modules/vehicles/mecha/mech_fabricator.dm [hunk 2]
```diff
@@ -149,7 +149,7 @@
 /obj/machinery/mecha_part_fabricator/emag_act(mob/user, obj/item/card/emag/emag_card)
 	if(obj_flags & EMAGGED)
 		return FALSE
-	if(user.job != JOB_ROBOTICIST)
+	if(!HAS_TRAIT(user, TRAIT_KNOW_ROBO_WIRES))
 		to_chat(user, span_warning("You clicking and typing but don’t understand what to do with it"))
```
Comment (MrMelbert) [inline code/modules/vehicles/mecha/mech_fabricator.dm:153]: Can we update this message as well? it doesn't say much
```suggestion
		to_chat(user, span_warning("You're unsure about [emag_card ? "where to swipe [emag_card] over" : "how to override"] [src] for any effect. Maybe if you had more knowledge of robotics..."))

```

File: code/modules/vehicles/mecha/mech_fabricator.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/vehicles/mecha/mech_fabricator.dm]: In either the discord or the absolute mess that is the PR that added the Justice mech.

**Pull request merged**